### ISSUE TITLE
 mapDispatchToProps-prefer-object triggering false positive on shorthand methods

### ISSUE THREAD

**Issue by calinoracation** [DianaSuvorova/eslint-plugin-react-redux#10]
One of the common errors I'm seeing is this rule throwing when we have a piece of code like the below that is returning an object:

```jsx
const mapDispatchToProps = dispatch => ({
  onDoSomething: () => dispatch(toSomethingElse()),
});
```

**Comment by DianaSuvorova** [comment]
My assumption was that one shouldn't define any special logic inside `mapDispatchToProps`, and that in above example you would need to create another action instead (ie`onDoSomething `).
But I can see how this can be overly strict for some projects. Let me look into relaxing this rule to accommodate for the use case.

**Comment by DianaSuvorova** [comment]
I think this rule needs to evolve into 2 independent checks instead:

1. mapDispatchToProps-no-bindActionCreators
Even though the react-redux  [docs](https://github.com/reactjs/react-redux/blob/master/docs/api.md#examples) provide tons of example with bindActionCreators, there should be no need for it and it may be rather confusing:

> Normally you should just call dispatch directly on your Store instance. If you use Redux with React, react-redux will provide you with the dispatch function so you can call it directly, too.

> The only use case for bindActionCreators is when you want to pass some action creators down to a component that isn't aware of Redux, and you don't want to pass dispatch or the Redux store to it.

[source](https://redux.js.org/docs/api/bindActionCreators.html)

2. mapDispatchToProps-consistent-dispatch
This rule would warn the following case:

```js
const mapDispatchToProps = dispatch => ({
  onDoSomething: () => dispatch(toSomethingElse()),
  action1: action1
});
```
while below two are correct:

```js
const mapDispatchToProps = dispatch => ({
  onDoSomething: () => dispatch(toSomethingElse()),
  action1: dispatch(action1())
});
```

```js
const mapDispatchToProps = {
  onDoSomething: doSomethingElse
  action1,
});
```
Let me know your thoughts

**Comment by DianaSuvorova** [comment]
Oh, and one more rule:

3. mapDispatchToProps-prefer-shorthand.

This would warn:

```js
const mapDispatchToProps = dispatch => ({
  onDoSomething: () => dispatch(onDoSomething()),
  action1: () => dispatch(action1()),
  action2: (arg1, arg2) => dispatch(action2(arg1, arg2)),
});
```

because this can be affectively rewritten as:

```js
const mapDispatchToProps = {
  onDoSomething,
  action1,
  action2,
});
```

and If there is any logic , the rule will consider the usage correct

```js
const mapDispatchToProps = dispatch => ({
  onDoSomething: () => dispatch(onDoSomethingElse()),
  action1: () => dispatch(action1()),
  action2: (arg1, arg2) => dispatch(action2(arg1 + arg2)),
});
```

**Comment by DianaSuvorova** [comment]
Will close this. Created #24 for the only remaining action item.
